sqlmesh/dbt/model.py

```diff
@@ -358,14 +364,22 @@ def sqlmesh_config_fields(self) -> t.Set[str]:
 
     def to_sqlmesh(self, context: DbtContext) -> Model:
         """Converts the dbt model into a SQLMesh model."""
-        dialect = context.dialect
+        model_dialect = self.dialect or context.default_dialect
         query = d.jinja_query(self.sql_no_config)
 
         optional_kwargs: t.Dict[str, t.Any] = {}
 
         if self.partition_by:
             optional_kwargs["partitioned_by"] = (
-                [exp.to_column(val) for val in self.partition_by]
+                [
+                    exp.to_column(
+                        t.cast(
+                            exp.Column, d.parse_one(val, dialect=model_dialect, into=exp.Column)
+                        ),
+                        dialect=context.target.dialect,
+                    )
+                    for val in self.partition_by
+                ]
```

Comment:
I'm not sure this is needed. Aren't we creating directly into the AST and the sqlglot generator will handle transpiling for us? How is `partitioned_by` handled on the sqlmesh side?

Reply:
`exp.to_column(val)` is parsing the raw column string from the model definition and returning an expression. If someone includes dialect-specific characters, I think it will choke. 

cluster_by is also parsing column names just below and passes a dialect, so maybe it not being here already is an oversight?
https://github.com/TobikoData/sqlmesh/blob/b7c659e66b6eab5ec30a6f5d341937d8746de683/sqlmesh/dbt/model.py#L377

Either way, this initial implementation is messy, so I'll clean it up and we can remove if it's not necessary.